internal/tofu/eval_context_builtin.go

```diff
@@ -309,12 +310,17 @@ func (ctx *BuiltinEvalContext) EvaluateBlock(body hcl.Body, schema *configschema
 	diags = diags.Append(evalDiags)
 	val, evalDiags := scope.EvalBlock(body, schema)
 	diags = diags.Append(evalDiags)
+	val, depDiags := marks.DeprecatedDiagnosticsInBody(val, body)
+	diags = diags.Append(depDiags)
 	return val, body, diags
 }
 
 func (ctx *BuiltinEvalContext) EvaluateExpr(expr hcl.Expression, wantType cty.Type, self addrs.Referenceable) (cty.Value, tfdiags.Diagnostics) {
 	scope := ctx.EvaluationScope(self, nil, EvalDataForNoInstanceKey)
-	return scope.EvalExpr(expr, wantType)
+	v, diags := scope.EvalExpr(expr, wantType)
+	v, depDiags := marks.DeprecatedDiagnosticsInExpr(v, expr)
```

Comment:
Can we move this call to scope.EvalExpr?

Reply:
[46fc042](https://github.com/opentofu/opentofu/pull/2633/commits/46fc042f1b327b00d1eb0715cdedae7160ab8e01)